src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -10946,5 +10966,1580 @@ unsafe class Test
+                //     C6 : I1
+                Diagnostic(ErrorCode.ERR_UnimplementedInterfaceMember, "I1").WithArguments("C6", "I1.M01()").WithLocation(36, 10),
+                // (38,12): error CS0539: 'C6.M01()' in explicit interface declaration is not found among members of the interface that can be implemented
+                //     int I1.M01() => throw null;
+                Diagnostic(ErrorCode.ERR_InterfaceMemberNotFound, "M01").WithArguments("C6.M01()").WithLocation(38, 12)
+                );
+        }
+
+        [Fact]
+        public void ImplementAbstractStaticMethod_03()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    abstract static void M01();
+}
+
+interface I2 : I1
+{}
+
+interface I3 : I1
+{
+    public virtual void M01() {}
+}
+
+interface I4 : I1
+{
+    static void M01() {}
+}
+
+interface I5 : I1
+{
+    void I1.M01() {}
+}
+
+interface I6 : I1
+{
+    static void I1.M01() {}
+}
+
+interface I7 : I1
+{
+    abstract static void M01();
+}
+";
+
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            compilation1.VerifyDiagnostics(
+                // (12,25): warning CS0108: 'I3.M01()' hides inherited member 'I1.M01()'. Use the new keyword if hiding was intended.
+                //     public virtual void M01() {}
+                Diagnostic(ErrorCode.WRN_NewRequired, "M01").WithArguments("I3.M01()", "I1.M01()").WithLocation(12, 25),
+                // (17,17): warning CS0108: 'I4.M01()' hides inherited member 'I1.M01()'. Use the new keyword if hiding was intended.
+                //     static void M01() {}
+                Diagnostic(ErrorCode.WRN_NewRequired, "M01").WithArguments("I4.M01()", "I1.M01()").WithLocation(17, 17),
+                // (22,13): error CS0539: 'I5.M01()' in explicit interface declaration is not found among members of the interface that can be implemented
+                //     void I1.M01() {}
+                Diagnostic(ErrorCode.ERR_InterfaceMemberNotFound, "M01").WithArguments("I5.M01()").WithLocation(22, 13),
+                // (27,20): error CS0106: The modifier 'static' is not valid for this item
+                //     static void I1.M01() {}
+                Diagnostic(ErrorCode.ERR_BadMemberFlag, "M01").WithArguments("static").WithLocation(27, 20),
+                // (27,20): error CS0539: 'I6.M01()' in explicit interface declaration is not found among members of the interface that can be implemented
+                //     static void I1.M01() {}
+                Diagnostic(ErrorCode.ERR_InterfaceMemberNotFound, "M01").WithArguments("I6.M01()").WithLocation(27, 20),
+                // (32,26): warning CS0108: 'I7.M01()' hides inherited member 'I1.M01()'. Use the new keyword if hiding was intended.
```

Comment:
If this isn't marked as `newslot`, how does the runtime know this hides the base version? Consider also verifying the `newslot` behavior of this method.

Reply:
This is probably me just not understanding part of the spec, I guess: I would have expected that the runtime would be updated so `newslot` could apply to static methods, so that a future world where you can say `static abstract new void M1();` would someday exist.

---
In reply to: [623509643](https://github.com/dotnet/roslyn/pull/52969#discussion_r623509643) [](ancestors = 623509643)

Reply:
> If this isn't marked as newslot, how does the runtime know this hides the base version?

This is how static methods work, this is part of runtime specification. New slot is set to specify that a virtual method is not an override, static methods never override anything.

> Consider also verifying the newslot behavior of this method.

DefineAbstractStaticMethod_01 already covers that.

Reply:
> so that a future world where you can say `static abstract new void M1();` would someday exist.

ie, something like this:

```cs
interface I1
{
    static abstract void M1();
}
interface I1 : I2
{
    static void I1.M1() {}
    static abstract new void M1();
}
```

---
In reply to: [623522140](https://github.com/dotnet/roslyn/pull/52969#discussion_r623522140) [](ancestors = 623522140)

Reply:
> so that a future world where you can say static abstract new void M1(); would someday exist.

This future is already here. It exists. This is a warning, not an error.